### PR TITLE
Add mapper object

### PR DESCRIPTION
**Pull request by garunagiri**
Added a new mapper object to be used for parsers. This object would hold all information under a parser in the yaml file instead of a relationship object. This should allow the addition of new elements easily in the future. Parsers now use this object.

### CODE REVIEW

File: app/objects/c_mapper.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from app.utility.base_object import BaseObject
+
+
+class Mapper(BaseObject):
+
+    @property
+    def unique(self):
+        return '%s%s%s%s' % (self.source, self.edge, self.target, self.json_key)
```
Comment (wbooth) [inline app/objects/c_mapper.py]: not every mapper is going to have a json key. Here I would think about allowing any number of arbitrary properties to be added.

Comment (garunagiri) [reply app/objects/c_mapper.py]: I imagined additional properties could be added specifically to the Mapper object and other places, but that's more work than needed. I'll change the object to have a 'misc' dict or some other name.

**Comment by garunagiri** [comment]
@wbooth modified to allow for any number of properties to be added easily.

File: app/objects/c_mapper.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+from app.utility.base_object import BaseObject
+
+
+class Mapper(BaseObject):
+
+    @property
+    def unique(self):
+        uniq = ""
```
Comment (privateducky) [inline app/objects/c_mapper.py]: use single quotes in CALDERA code

**Comment by wbooth** [comment]
closing given https://github.com/mitre/caldera/pull/768

**Pull request closed without merging**